plugins/module_utils/foreman_helper.py

```diff
@@ -737,7 +764,7 @@ def _update_entity(self, resource, desired_entity, current_entity, params, forem
                 old_value = sorted(old_value, key=operator.itemgetter(sort_key))
             if new_value != old_value:
                 payload[key] = value
-        if payload:
+        if self._validate_supported_payload(resource, 'update', payload):
```

Comment:
Should we not sort the parameters before comparing?
That way it would get idempotent in cases where you define a parameter that does not exist.
Or is it the entity itself, that is purged from `payload`? In that case i expect the warning to emit that the entity itself were unsupported.

Reply:
not sure what you mean, I didn't touch any sorting-related things here?

Reply:
Sorry, i didn't mean sorting, but filtering. So if we filtered the desired parameters by what is applicable upstream, the comparing code should not trigger an update anymore.
The comment about the warning is probably wrong. I was worried about cases where payload was of the form
```
{"host": {"unsupported": "spooky"}}
```
but it's probably never nested like this.

Reply:
Yeah, but it doesn't work like this :see_no_evil: 

My validator (ab)uses `_resource_prepare_params` which prepares an API request, most importantly the *input* is a flat dict but the output (can be) a dict-of-dicts. So the output can't be (easily) used for comparison of individual parameters anymore.

And now that I've written that, I notice that my warning code relies on the fact it's not like that. Damn.